List of quarterlies screen redesign

TinasheMzondiwa opened the pull request:
# What did you change:

_Enter a brief description so PR reviewers know what to expect_

# Screenshot/GIFs of this change

| Light                        | Dark                        |
| ----------------------------- | ---------------------------- |
| ![Light](https://user-images.githubusercontent.com/5994683/130340100-9c398e75-1858-4b4c-a91e-98cddad3d8cf.png)  | ![Dark](https://user-images.githubusercontent.com/5994683/130340108-99808268-4a30-4422-835e-bc21ad2dadcd.png)

![Screenshot_1629599841](https://user-images.githubusercontent.com/5994683/130340120-26b4ba29-2928-4aa7-92a6-0ea3a8a7253a.png)


# Merge checklist

- [x] Automated (unit/ui) tests
- [x] Manual tests

TinasheMzondiwa committed:
Use surface color

TinasheMzondiwa committed:
Refactor quarterlies appbar into component

TinasheMzondiwa committed:
Refresh quarterlies when language code changes

TinasheMzondiwa committed:
Filter and group quarterlies

TinasheMzondiwa committed:
Set theme color in SSPrefs

TinasheMzondiwa committed:
Show grouped and list quarterlies

TinasheMzondiwa committed:
Use minHeight for title

TinasheMzondiwa committed:
Remove language prompt

TinasheMzondiwa committed:
Revert livedata to flow

TinasheMzondiwa committed:
Add click callbacks

TinasheMzondiwa committed:
Add ComposeRecyclerViewUtils.kt

TinasheMzondiwa committed:
Add Activity transition

TinasheMzondiwa committed:
Remove legacy xml layouts

TinasheMzondiwa committed:
Show QuarterliesActivity by group

TinasheMzondiwa committed:
Use separate adapters for list and group

TinasheMzondiwa committed:
Add translations

TinasheMzondiwa committed:
Add tests

TinasheMzondiwa committed:
Merge branch 'develop' into tm/feat/quarterly_list

VitalikL commented:
@TinasheMzondiwa Great work!

Tested it for the first time this morning. There are few minor observations:

1. The `color_primary_dark` is still being present (example About Screen or Read Screen)
2. For tablets and larger screens please increase item spacing and side margins (see example of the iPad implementation)

![IMG_BABB2B91FD61-1](https://user-images.githubusercontent.com/4108258/130244936-9fd70d5a-8c5f-4ebe-b11d-93382cc0b6d0.jpeg)

3. Slightly reduce the gray color of the bottom gradient of the horizontal group.

TinasheMzondiwa committed:
Provide configurable dimens

TinasheMzondiwa committed:
Adjust for large screen sizes

TinasheMzondiwa commented:
> @TinasheMzondiwa Great work!
> 
> Tested it for the first time this morning. There are few minor observations:
> 
> 1. The `color_primary_dark` is still being present (example About Screen or Read Screen)
> 2. For tablets and larger screens please increase item spacing and side margins (see example of the iPad implementation)
> 
> ![IMG_BABB2B91FD61-1](https://user-images.githubusercontent.com/4108258/130244936-9fd70d5a-8c5f-4ebe-b11d-93382cc0b6d0.jpeg)
> 
> 1. Slightly reduce the gray color of the bottom gradient of the horizontal group.

@VitalikL I have addressed the issues.

1. I will open a new pr with this. This one is already too big
2. Fixed!
3. Fixed!

VitalikL reviewed (approved):
@TinasheMzondiwa 

Can you align the group name (ie. Standard Adult, Inverse) and the quarterly title on the left with the image. Right now it seems there is a bit of a padding.

Otherwise, looks good!

TinasheMzondiwa committed:
Adjust content padding
